test_util/test_runner.py

```diff
@@ -64,55 +56,29 @@ def integration_test(
         for key, value in add_env.items():
             extra_env = key + '=' + value
             test_env.append(extra_env)
-    test_env = ['export '+e+'\n' for e in test_env]
-    test_env = ''.join(test_env)
-    test_cmd = 'py.test -vv ' + ci_flags + ' /integration_test.py'
-
-    log.info('Building py.test image')
-    # Make a clean docker context
-    temp_dir = tempfile.mkdtemp()
-    cmd_script = """
-#!/bin/bash
-set -euo pipefail; set -x
-{test_env}
-{test_cmd}
-""".format(test_env=test_env, test_cmd=test_cmd)
-    write_string(join(temp_dir, 'test_wrapper.sh'), cmd_script)
-    check_call(['cp', test_script, join(temp_dir, 'integration_test.py')])
-    check_call(['cp', pytest_docker, join(temp_dir, 'Dockerfile')])
-    check_call(['cp', test_server, join(temp_dir, 'test_server.py')])
-    check_call(['cp', test_server_docker, join(temp_dir, 'test_server_Dockerfile')])
-    check_call(['docker', 'build', '-t', 'py.test', temp_dir])
 
-    log.info('Exporting py.test image')
-    pytest_image_tar = 'DCOS_integration_test.tar'
-    check_call(['docker', 'save', '-o', join(temp_dir, pytest_image_tar), 'py.test'])
-
-    log.info('Transferring py.test image')
-    tunnel.write_to_remote(join(temp_dir, pytest_image_tar), join(test_dir, pytest_image_tar))
-    log.info('Loading py.test image on remote host')
-    tunnel.remote_cmd(['docker', 'load', '-i', join(test_dir, pytest_image_tar)])
+    test_wrapper = """#!/bin/bash
+{env}
+cd /opt/mesosphere/active/dcos-integration-test
+py.test -vv {ci_flags}
+"""
+    test_env_str = ''.join(['export '+e+'\n' for e in test_env])
+    write_string('test_wrapper.sh', test_wrapper.format(env=test_env_str, ci_flags=ci_flags))
 
-    test_container_name = 'int_test_' + str(int(time.time()))
-    docker_cmd = ['docker', 'run', '--net=host', '--name='+test_container_name, 'py.test']
     try:
+        wrapper_path = join(test_dir, 'test_wrapper.sh')
         log.info('Running integration test...')
-        tunnel.remote_cmd(docker_cmd, timeout=timeout)
+        tunnel.write_to_remote('test_wrapper.sh', wrapper_path)
+        dump_logs(tunnel.remote_cmd(['bash', wrapper_path], timeout=timeout))
         log.info('Successful test run!')
-    except CalledProcessError:
+    except CalledProcessError as e:
         log.exception('Test failed!')
+        dump_logs(e.output)
```

Comment:
It should be possible to do either of two things.
1. Read and write the data stream as bytes (best option), never decoding it just forwarding.
2. Readn and write the data stream as utf-8, and make sure to set PYTHONENCODING to `utf-8` in the environment, then write the data back out as utf-8.